pyfive/inspect.py

```diff
@@ -0,0 +1,201 @@
+            edim = (obj.dims[axis][0].name.split('/')[-1], size)
+
+        elif size in real_dimensions.values():
+            dim_name = next(name for name, sz in real_dimensions.items() if sz == size)
+            edim = (dim_name, size)
+        else:
+            # make or reuse a phony dimension name
+            if size not in phonys:
+                phonys[size] = f"phony_dim_{len(phonys)}"
+            pname = phonys[size]
+            edim = (pname,size)
+ 
+        obj.__inspected_dims.append(edim)
+        alldims.add(edim)
+
+    return obj, alldims, phonys
+
+
+def dump_header(obj, indent, real_dimensions, special):
+    """ Pretty print a group within an HDF5 file (including the root group) """
+
+    def printattr(attrs, ommit=[]):
+        """ Pretty print a set of attributes """
+        for k,v in attrs.items():
+            if k not in ommit:
+                if isinstance(v, bytes):
+                    v = f'"{v.decode("utf-8")}"'
+                elif isinstance(v,str):
+                    v = f'"{v}"'
+                safe_print(f"{indent}{dindent}{dindent}{name}:{k} = {v} ;")
+
+    dims = set()
+    datasets = {}
+    groups = {}
+    phonys = {}
+
+    for name in obj:
+        item = obj.get_lazy_view(name)
+        if isinstance(item, Dataset):
+            if str(item.attrs.get('NAME','None')).startswith('This is a netCDF dimension but not a'):
+                    continue
+            datasets[name]=item
+        elif isinstance(item, Group):
+            groups[name]=item
+
+
+    for ds in datasets.values():
+        ds, dims, phonys = gather_dimensions(ds, dims, phonys, real_dimensions)
+    if dims:
+        safe_print(f"{indent}dimensions:")
+    dindent = '        '
+    for d in dims:
+        safe_print(f'{indent}{dindent}{d[0]} = {d[1]};')
+    
+    print(f"{indent}variables:")
+    for name,ds in datasets.items():
+        
+        # Variable type
+        dtype_str = clean_types(ds.dtype)
+
+        # Dimensions for this variable (use dims if available)
+        if hasattr(ds,'__inspected_dims'):
+            dim_names = [d[0] for d in ds.__inspected_dims]
+            dim_str = "(" + ", ".join(dim_names) + ")" if dim_names else ""
+            safe_print(f"{indent}{dindent}{dtype_str} {name}{dim_str} ;")
+
+        # Attributes
+        ommit = ['CLASS','NAME','_Netcdf4Dimid',
+                 'REFERENCE_LIST','DIMENSION_LIST','DIMENSION_LABELS','_Netcdf4Coordinates']
+        
+        printattr(ds.attrs, ommit)
+
+        if special:
+            extras = {'_Storage':{0:'Compact',1:'Contiguous',2:'Chunked'}[ds.id.layout_class]}
+            if ds.id.layout_class==2:
+                extras['_n_chunks'] = ds.id.get_num_chunks()
+                if extras['_n_chunks'] != 0:
+                    extras['_chunk_shape'] = ds.id.chunks
+                    extras['_btree_range'] = ds.id.btree_range
+                    extras['_first_chunk'] = ds.id.get_chunk_info(0).byte_offset
```

Comment:
Given the new `first_chunk` property I suggested in `h5d.py`:
```suggestion
                    extras['_first_chunk'] = ds.id.first_chunk
```